### ISSUE TITLE
Can't run with Google Web Toolkit 2.6 RC1

### ISSUE THREAD

**Issue by GoogleCodeExporter** [anne/google-gin#197]
```
What steps will reproduce the problem?
1. Create a program using google-gin.
2. Run in eclipse with Google Plugin for Eclipse.

What is the expected output? What do you see instead?
- Expected: An instance of an interface with extends Ginjector is instantiated 
and its method returns an object.
- Actual: "java.lang.RuntimeException: Deferred binding failed for 
'tv.dyndns.kishibe.gin.client.ExperimentalGinjector' (did you forget to inherit 
a required module?)" is shown.

What version of the product are you using? On what operating system?
- google-gin 2.1.1
- Google Web Toolkit 2.6 RC1

Please provide any additional information below.

The whole error message was below:
========================================
17:15:56.246 [ERROR] [ginexperiment] Failed to create an instance of 
'tv.dyndns.kishibe.gin.client.GinExperiment' via deferred binding 

java.lang.RuntimeException: Deferred binding failed for 
'tv.dyndns.kishibe.gin.client.ExperimentalGinjector' (did you forget to inherit 
a required module?)
    at com.google.gwt.dev.shell.GWTBridgeImpl.create(GWTBridgeImpl.java:53)
    at com.google.gwt.core.shared.GWT.create(GWT.java:72)
    at com.google.gwt.core.client.GWT.create(GWT.java:86)
    at tv.dyndns.kishibe.gin.client.GinExperiment.&lt;init&gt;(GinExperiment.java:8)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
    at java.lang.reflect.Constructor.newInstance(Unknown Source)
    at com.google.gwt.dev.shell.ModuleSpace.rebindAndCreate(ModuleSpace.java:480)
    at com.google.gwt.dev.shell.ModuleSpace.onLoad(ModuleSpace.java:390)
    at com.google.gwt.dev.shell.OophmSessionHandler.loadModule(OophmSessionHandler.java:200)
    at com.google.gwt.dev.shell.BrowserChannelServer.processConnection(BrowserChannelServer.java:526)
    at com.google.gwt.dev.shell.BrowserChannelServer.run(BrowserChannelServer.java:364)
    at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NoClassDefFoundError: com/google/gwt/dev/util/Preconditions
    at com.google.gwt.inject.rebind.reflect.MethodLiteral.get(MethodLiteral.java:59)
    at com.google.gwt.inject.rebind.util.MemberCollector.getTypeMethods(MemberCollector.java:311)
    at com.google.gwt.inject.rebind.util.MemberCollector.accumulateMembers(MemberCollector.java:269)
    at com.google.gwt.inject.rebind.util.MemberCollector.collect(MemberCollector.java:255)
    at com.google.gwt.inject.rebind.util.MemberCollector.getMethods(MemberCollector.java:227)
    at com.google.gwt.inject.rebind.BindingsProcessor.validateMethods(BindingsProcessor.java:167)
    at com.google.gwt.inject.rebind.BindingsProcessor.process(BindingsProcessor.java:97)
    at com.google.gwt.inject.rebind.GinjectorGeneratorImpl.generate(GinjectorGeneratorImpl.java:79)
    at com.google.gwt.inject.rebind.GinjectorGenerator.generate(GinjectorGenerator.java:74)
    at com.google.gwt.core.ext.IncrementalGenerator.generateNonIncrementally(IncrementalGenerator.java:40)
    at com.google.gwt.dev.javac.StandardGeneratorContext.runGeneratorIncrementally(StandardGeneratorContext.java:676)
    at com.google.gwt.dev.cfg.RuleGenerateWith.realize(RuleGenerateWith.java:41)
    at com.google.gwt.dev.shell.StandardRebindOracle$Rebinder.rebind(StandardRebindOracle.java:79)
    at com.google.gwt.dev.shell.StandardRebindOracle.rebind(StandardRebindOracle.java:276)
    at com.google.gwt.dev.shell.ShellModuleSpaceHost.rebind(ShellModuleSpaceHost.java:147)
    at com.google.gwt.dev.shell.ModuleSpace.rebind(ModuleSpace.java:600)
    at com.google.gwt.dev.shell.ModuleSpace.rebindAndCreate(ModuleSpace.java:470)
    at com.google.gwt.dev.shell.GWTBridgeImpl.create(GWTBridgeImpl.java:49)
    at com.google.gwt.core.shared.GWT.create(GWT.java:72)
    at com.google.gwt.core.client.GWT.create(GWT.java:86)
    at tv.dyndns.kishibe.gin.client.GinExperiment.&lt;init&gt;(GinExperiment.java:8)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
    at java.lang.reflect.Constructor.newInstance(Unknown Source)
    at com.google.gwt.dev.shell.ModuleSpace.rebindAndCreate(ModuleSpace.java:480)
    at com.google.gwt.dev.shell.ModuleSpace.onLoad(ModuleSpace.java:390)
    at com.google.gwt.dev.shell.OophmSessionHandler.loadModule(OophmSessionHandler.java:200)
    at com.google.gwt.dev.shell.BrowserChannelServer.processConnection(BrowserChannelServer.java:526)
    at com.google.gwt.dev.shell.BrowserChannelServer.run(BrowserChannelServer.java:364)
    at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.ClassNotFoundException: 
com.google.gwt.dev.util.Preconditions
    at java.net.URLClassLoader$1.run(Unknown Source)
    at java.net.URLClassLoader$1.run(Unknown Source)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(Unknown Source)
    at java.lang.ClassLoader.loadClass(Unknown Source)
    at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
    at java.lang.ClassLoader.loadClass(Unknown Source)
    at com.google.gwt.inject.rebind.reflect.MethodLiteral.get(MethodLiteral.java:59)
    at com.google.gwt.inject.rebind.util.MemberCollector.getTypeMethods(MemberCollector.java:311)
    at com.google.gwt.inject.rebind.util.MemberCollector.accumulateMembers(MemberCollector.java:269)
    at com.google.gwt.inject.rebind.util.MemberCollector.collect(MemberCollector.java:255)
    at com.google.gwt.inject.rebind.util.MemberCollector.getMethods(MemberCollector.java:227)
    at com.google.gwt.inject.rebind.BindingsProcessor.validateMethods(BindingsProcessor.java:167)
    at com.google.gwt.inject.rebind.BindingsProcessor.process(BindingsProcessor.java:97)
    at com.google.gwt.inject.rebind.GinjectorGeneratorImpl.generate(GinjectorGeneratorImpl.java:79)
    at com.google.gwt.inject.rebind.GinjectorGenerator.generate(GinjectorGenerator.java:74)
    at com.google.gwt.core.ext.IncrementalGenerator.generateNonIncrementally(IncrementalGenerator.java:40)
    at com.google.gwt.dev.javac.StandardGeneratorContext.runGeneratorIncrementally(StandardGeneratorContext.java:676)
    at com.google.gwt.dev.cfg.RuleGenerateWith.realize(RuleGenerateWith.java:41)
    at com.google.gwt.dev.shell.StandardRebindOracle$Rebinder.rebind(StandardRebindOracle.java:79)
    at com.google.gwt.dev.shell.StandardRebindOracle.rebind(StandardRebindOracle.java:276)
    at com.google.gwt.dev.shell.ShellModuleSpaceHost.rebind(ShellModuleSpaceHost.java:147)
    at com.google.gwt.dev.shell.ModuleSpace.rebind(ModuleSpace.java:600)
    at com.google.gwt.dev.shell.ModuleSpace.rebindAndCreate(ModuleSpace.java:470)
    at com.google.gwt.dev.shell.GWTBridgeImpl.create(GWTBridgeImpl.java:49)
    at com.google.gwt.core.shared.GWT.create(GWT.java:72)
    at com.google.gwt.core.client.GWT.create(GWT.java:86)
    at tv.dyndns.kishibe.gin.client.GinExperiment.&lt;init&gt;(GinExperiment.java:8)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
    at java.lang.reflect.Constructor.newInstance(Unknown Source)
    at com.google.gwt.dev.shell.ModuleSpace.rebindAndCreate(ModuleSpace.java:480)
    at com.google.gwt.dev.shell.ModuleSpace.onLoad(ModuleSpace.java:390)
    at com.google.gwt.dev.shell.OophmSessionHandler.loadModule(OophmSessionHandler.java:200)
    at com.google.gwt.dev.shell.BrowserChannelServer.processConnection(BrowserChannelServer.java:526)
    at com.google.gwt.dev.shell.BrowserChannelServer.run(BrowserChannelServer.java:364)
    at java.lang.Thread.run(Unknown Source)
========================================

The minimum code set is attached.

```

Original issue reported on code.google.com by `hn...@google.com` on 18 Nov 2013 at 2:01
- Merged into: #195

Attachments:
- [GinExperiment.zip](https://storage.googleapis.com/google-code-attachments/google-gin/issue-197/comment-0/GinExperiment.zip)

**Comment by GoogleCodeExporter** [comment]
```
This is a known issue (GIN was using internal APIs of GWT that have been 
removed), and has been fixed in trunk already (r278)
```

Original comment by `t.broyer` on 18 Nov 2013 at 6:13

**Comment by GoogleCodeExporter** [comment]
```
Thank you for responding!  Please close this issue.

Thanks,

```

Original comment by `hn...@google.com` on 18 Nov 2013 at 6:30

**Comment by GoogleCodeExporter** [comment]
Original comment by `aragos` on 18 Nov 2013 at 6:06
- Changed state: **Duplicate**

**Comment by GoogleCodeExporter** [comment]
```
I also get this issue when using EventsBinder (with 2.6.0-rc1 and Java 7 on OSX 
Mavericks)

[INFO] Caused by: java.lang.NoClassDefFoundError: 
com/google/gwt/dev/util/Preconditions
     [exec] [INFO]  at com.google.web.bindery.event.gwt.rebind.binder.EventBinderGenerator.getTargetType(EventBinderGenerator.java:73)
     [exec] [INFO]  at com.google.web.bindery.event.gwt.rebind.binder.EventBinderGenerator.generate(EventBinderGenerator.java:55)
     [exec] [INFO]  at com.google.gwt.core.ext.IncrementalGenerator.generateNonIncrementally(IncrementalGenerator.java:40)
     [exec] [INFO]  at com.google.gwt.dev.javac.StandardGeneratorContext.runGeneratorIncrementally(StandardGeneratorContext.java:676)
     [exec] [INFO]  at com.google.gwt.dev.cfg.RuleGenerateWith.realize(RuleGenerateWith.java:41)
     [exec] [INFO]  at com.google.gwt.dev.shell.StandardRebindOracle$Rebinder.rebind(StandardRebindOracle.java:79)
     [exec] [INFO]  at com.google.gwt.dev.shell.StandardRebindOracle.rebind(StandardRebindOracle.java:276)
     [exec] [INFO]  at com.google.gwt.dev.shell.StandardRebindOracle.rebind(StandardRebindOracle.java:265)
     [exec] [INFO]  at com.google.gwt.dev.DistillerRebindPermutationOracle.getAllPossibleRebindAnswers(DistillerRebindPermutationOracle.java:91)
     [exec] [INFO]  at com.google.gwt.dev.jjs.impl.UnifyAst$UnifyVisitor.handleGwtCreate(UnifyAst.java:387)
     [exec] [INFO]  at com.google.gwt.dev.jjs.impl.UnifyAst$UnifyVisitor.handleMagicMethodCall(UnifyAst.java:465)
     [exec] [INFO]  at com.google.gwt.dev.jjs.impl.UnifyAst$UnifyVisitor.endVisit(UnifyAst.java:255)
     [exec] [INFO]  at com.google.gwt.dev.jjs.ast.JMethodCall.traverse(JMethodCall.java:244)
     [exec] [INFO]  at com.google.gwt.dev.jjs.ast.JModVisitor.traverse(JModVisitor.java:361)
     [exec] [INFO]  at com.google.gwt.dev.jjs.ast.JModVisitor.accept(JModVisitor.java:273)
     [exec] [INFO]  ... 37 more
     [exec] [INFO] Caused by: java.lang.ClassNotFoundException: com.google.gwt.dev.util.Preconditions
     [exec] [INFO]  at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
     [exec] [INFO]  at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
     [exec] [INFO]  at java.security.AccessController.doPrivileged(Native Method)
     [exec] [INFO]  at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
     [exec] [INFO]  at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
     [exec] [INFO]  at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
     [exec] [INFO]  at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
     [exec] [INFO]  ... 52 more
     [exec] [INFO]       [ERROR] at MGWTPlaceHistoryHandler.java(222): GWT.create(MGWTPlaceHistoryHandler$EventsBinder.class)
     [exec] [INFO]          com.google.gwt.dev.jjs.ast.JMethodCall
     [exec] [INFO]       [ERROR] at MGWTPlaceHistoryHandler.java(222): (MGWTPlaceHistoryHandler$EventsBinder) GWT.create(MGWTPlaceHistoryHandler$EventsBinder.class)
     [exec] [INFO]          com.google.gwt.dev.jjs.ast.JCastOperation
     [exec] [INFO]       [ERROR] at MGWTPlaceHistoryHandler.java(222): final MGWTPlaceHistoryHandler$EventsBinder eventBinder = (MGWTPlaceHistoryHandler$EventsBinder) GWT.create(MGWTPlaceHistoryHandler$EventsBinder.class)
     [exec] [INFO]          com.google.gwt.dev.jjs.ast.JDeclarationStatement
     [exec] [INFO]       [ERROR] at MGWTPlaceHistoryHandler.java(37): {
     [exec] [INFO]   final MGWTPlaceHistoryHandler$EventsBinder eventBinder = (MGWTPlaceHistoryHandler$EventsBinder) GWT.create(MGWTPlaceHistoryHandler$EventsBinder.class);
     [exec] [INFO] }
     [exec] [INFO]          com.google.gwt.dev.jjs.ast.JBlock
     [exec] [INFO]       [ERROR] at MGWTPlaceHistoryHandler.java(37): {
     [exec] [INFO]   final MGWTPlaceHistoryHandler$EventsBinder eventBinder = (MGWTPlaceHistoryHandler$EventsBinder) GWT.create(MGWTPlaceHistoryHandler$EventsBinder.class);
     [exec] [INFO] }
     [exec] [INFO]          com.google.gwt.dev.jjs.ast.JMethodBody
     [exec] [INFO]       [ERROR] at MGWTPlaceHistoryHandler.java(37): private final void $init();
     [exec] [INFO] 
     [exec] [INFO]          com.google.gwt.dev.jjs.ast.JMethod
     [exec] [INFO] ------------------------------------------------------------------------
     [exec] [INFO] BUILD FAILURE
     [exec] [INFO] ------------------------------------------------------------------------
     [exec] [INFO] Total time: 1:04.267s
     [exec] [INFO] Finished at: Sun Nov 24 21:55:03 GMT+10:00 2013
     [exec] [INFO] Final Memory: 10M/81M
```

Original comment by `portleta...@gmail.com` on 24 Nov 2013 at 12:05

**Comment by GoogleCodeExporter** [comment]
```
Please contact the event binder project about this problem: 
https://github.com/google/gwteventbinder
```

Original comment by `aragos` on 24 Nov 2013 at 6:14

**Comment by GoogleCodeExporter** [comment]
```
Was solved in gwteventbinder 1.0.1.
```

Original comment by `gardella...@gmail.com` on 28 Nov 2013 at 8:21

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the note on EventBinder 1.0.1. I needed to upgrade to both it and 
Gin 2.1.2 to get GWT 2.6 to build my project.

It seems like Gin 2.1.2 doesn't like GWT 2.5.1 though (Deferred binding failed 
for 'com.google.gwt.user.client.ui.impl.ClippedImageImpl$Template' (did you 
forget to inherit a required module?)).
```

Original comment by `Splak...@gmail.com` on 16 Jan 2014 at 10:01
